src/apply.ml

```diff
@@ -0,0 +1,31 @@
+open Basics
+open Console
+open Terms
+open Refine
+
+(** [nbArgsOf t] counts the number of arguments of the type [t], i.e.
+    it counts the number of Prod binders located at the front of [t]. *)
+let rec nbArgsOf : term -> int = fun t ->
+  match unfold t with
+  | Prod(_,b)   -> let (_,b) = Bindlib.unbind b in 1 + nbArgsOf b
```

Comment:
I'd write `1 + snd (Bindlib.unbind b)` here, it is slightly shorter. Also, It is better to align the arrows of the pattern matching for readability.